source/ConsolidateCalamariPackagesTask/Consolidate.cs

```diff
@@ -0,0 +1,180 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Octopus.Build.ConsolidateCalamariPackagesTask
+{
+    public class Consolidate : IDisposable
+    {
+        private readonly ILog log;
+        private readonly MD5 md5 = MD5.Create();
+
+        public Consolidate(ILog log)
+        {
+            this.log = log;
+        }
+        
+        public string AssemblyVersion { get; set; } = typeof(Consolidate).Assembly.GetName().Version.ToString();
+
+        public (bool result, string packageFileName) Execute(string outputDirectory, IReadOnlyList<PackageReference> packageReferences)
+        {
+            if (!Directory.Exists(outputDirectory))
+            {
+                log.Error($"The output directory {outputDirectory} does not exist");
+                return (false, null);
+            }
+
+            var packagesToScan = packageReferences
+                .Where(p => /*p.Name.StartsWith("Sashimi.") ||*/ p.Name.StartsWith("Calamari"))
```

Comment:
Is this comment going to stay? We'd want to be scanning Sashimi packages, yeah?

Reply:
It won't, I'm working on that part right now. This at least gets Consolidated package into Server without `Calamari.Terraform`.